### DISCUSSION TITLE
How can I diagnose my keymap not working?

### DISCUSSION THREAD

**Discussion by jokajak** [LazyVim/LazyVim#1130] [Q&A] [answered]
I'm trying to map `<C-b>` as part of my specs.nvim plugin definition but the keymap isn't being picked up. So far I've taken the following steps:

```shell
cat > plugins/noice.lua <<_EOF_
return {
  "folke/noice.nvim",
  keys = {
    -- disable Ctrl-b so it can be used for specs
    { "<c-b>", false },
  },
}
_EOF_
cat > plugins/specs.lua <<_EOF_
local specs = {
  "edluffy/specs.nvim",
  event = "BufRead",
  lazy = true,
  keys = {
    {
      "<C-b>",
      function()
        require("specs").show_specs()
      end,
      desc = "Highlight cursor location",
      silent = true,
    },
}
return specs
_EOF_
```

How can I diagnose the <C-b> keymap not working? I feel like I'm missing something obvious. The keymap does *not* show up when I use `:WhichKey`

Thanks in advance.

**Comment by dpetka2001** [comment, marked as answer]
Could you try adding `, mode = {"i", "n", "s"}` when you disable the keymap in noice spec?

**Reply by jokajak** [reply]
Yes, thank you so much!

**Comment by tf42src** [comment]
Seconding the last question of OP, are there instructions somewhere how to debug if a key mapping is not working?

Afaik the following key mappings exist:

1. "native" (neo)vim (e.g. `dd` - delete line)
2. lazyvim keymaps in `keymaps.lua`
3. as part of the init/setup of a plugin itself

My issue is as well that pressing <leader> (or `:WhichKey`) also does not show any `d` mapping. 

I am trying to map "dw" to "diw" and it doesn't work either.

Here's a short example of what I tried:

```
local keymap = vim.api.nvim_set_keymap  -- also tried: vim.keymap.set
local delkeymap = vim.keymap.del

delkeymap("n", "dw", { silent = true })
keymap("n", "dw", "diw", { noremap = true, silent = true, desc = "Delete word under cursor" })
```

**Reply by tf42src** [reply]
Alright, found my issue: I had a misconfigured  plugin, see [here](https://github.com/rcarriga/nvim-notify/issues/197#issuecomment-2143358841) which seemed to not load keymaps.
